### ISSUE TITLE
Do not put "vendor" in cache_dirs by default

### ISSUE THREAD

**Issue by mbrodala** [nanobox-io/nanobox-engine-php#84]
I think that the `vendor` directory should not be in `cache_dirs` by default. Here are my reasons:

1. You don't have access to vendor packages on the host system. This basically makes code inspection and especially autocompletion in an editor/IDE impossible. A workaround would be running `composer install` once more on the host system but that would defeat the purpose since you need to set up the correct PHP environment on your host system then. A different workaround would be setting a different `vendor-dir` but this shouldn't be necessary.
2. It is impossible to explicitly remove the directory from `cache_dirs` since additional entries are added only, they don't overwrite the defaults.
3. Contrary to the previous point it is insanely easy to explicitly add the directory to `cache_dirs` through the boxfile.

Hence my proposal to remove `vendor` from the `cache_dirs`. What do you think?

**Comment by tim-bec** [comment]
👍 Please please remove vendor from cache_dirs

**Comment by mbrodala** [comment]
@tim-bec As a short-term solution you can fork this repository and apply my change from #85. I did that in https://github.com/mbrodala/nanobox-engine-php for example. Afterwards simply [refer to your fork in your `boxfile.yaml`](https://docs.nanobox.io/boxfile/run-config/#using-non-nanobox-engines).

**Comment by sanderson** [comment]
I created a branch called `option/no-default-cache-dirs' that removes `vendor` as a default cache dir. Though I don't think we want to merge it into master at this point, for those who want to use it, you can specify the branch when declaring the engine in your `boxfile.yml`:

```yaml
run.config:
  engine: php#option/no-default-cache-dirs
```

**Comment by tim-bec** [comment]
@sanderson thanks for that! Will nanobox keep this branch in sync with master?

**Comment by sanderson** [comment]
@tim-bec we will try to 🙂. I get notices for merges into master, so when those come in, I'll be sure merge the changes from master into this branch.

**Comment by ijy** [comment]
@sanderson Any reason why this wont be merged into master? I literally just spent a day trying to figure out issues associated with this.

Kudos @mbrodala !

**Comment by danhunsaker** [comment]
Yes, it won't be merged into master because the default expectation is that you're using Composer. If you're not, you know you're not, and you simply say so explicitly by adding `#option/no-default-cache-dirs` to the `engine` name.

**Comment by ijy** [comment]
At the minute however I have an issue which sees my application erroring out using the default php engine. If I swap it to one which doesn't cache the vendor directory it works fine. That's the only change.

**Comment by danhunsaker** [comment]
Sounds like you might need to clear out the cache, then. Fastest way is to `nanobox destroy` the app. The next `nanobox run` will rebuild the app from scratch, and things should work correctly from that point.

If that doesn't help, then activate the `option` branch and enjoy.

**Comment by ijy** [comment]
Is there an advantage to caching the vendor dir though? If there isn't an easy method for cache busting and it gets stuck, as it seems to have in this case and has previously done on other cases, it adds a lot of confusion trying to figure out why it isn't working. How much of a difference does caching it have?
